patroni/postgresql/config.py

```diff
@@ -687,6 +687,12 @@ def record_missmatch(mtype):
 
         wanted_recovery_params = self.build_recovery_params(member)
         for param, value in self._current_recovery_params.items():
+            # Skip certain parameters defined in the included postgres config files
+            # if we know that they are not specified in the patroni configuration.
```

Comment:
but changes in these params may still require reload / restart. Are they skipped simply because they are not a part of Patroni-managed configuration ?

Reply:
They are skipped only when 1. Values are defined neither in patroni.yaml nor in DCS and 2. current values are coming not from expected places, i.e. `postgresql.conf` or `postgresql.auto.conf`.

That means user is managing them on they own via the `custom_conf` and therefore must do reload/restart.